### PR TITLE
OSX Fixes

### PR DESCRIPTION
**Pull request by isuruf**

### CODE REVIEW

**Comment by isuruf** [comment]
Hmm, gcc on osx doesn't like `-dynamiclib`

File: lib/CL/devices/pthread/pthread_scheduler.c [hunk 1]
```diff
@@ -143,7 +143,7 @@ pthread_scheduler_uninit ()
   for (i = 0; i < scheduler.num_threads; ++i)
     {
       pthread_join (scheduler.thread_pool[i].thread, NULL);
-      PTHREAD_DESTROY_LOCK (&scheduler.thread_pool[i].lock);
+      // PTHREAD_DESTROY_LOCK (&scheduler.thread_pool[i].lock);
```
Comment (pjaaskel) [inline lib/CL/devices/pthread/pthread_scheduler.c]: This looks suspicious. Why? Otherwise looks OK to me.

Comment (isuruf) [reply lib/CL/devices/pthread/pthread_scheduler.c]: Isn't this a duplicate of https://github.com/pocl/pocl/blob/d899c94eae97c2a33ad8c38e96b0fb7d7e6d5893/lib/CL/devices/pthread/pthread_scheduler.c#L579 ?

Comment (franz) [reply lib/CL/devices/pthread/pthread_scheduler.c]: @isuruf yeah it is a duplicate. But anyway there's more to fix here. Can you drop the locking commit  from the PR please, i'll take care of the locks. Thanks

**Comment by franz** [comment]
Thanks. Tested in Travis and the Clang build now works, but the GCC is still failing for some reason.

File: CMakeLists.txt [hunk 2]
```diff
@@ -698,7 +698,11 @@ if(UNIX)
     # MacOS ld outputs useless warnings like
     # ld: warning: -macosx_version_min not specificed, assuming 10.7
     # suppress them with -w.
-    set(DEFAULT_HOST_LD_FLAGS "-dylib -w -lm")
+    if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
+        set(DEFAULT_HOST_LD_FLAGS "-dynamiclib -w -lm")
+    else()
+        set(DEFAULT_HOST_LD_FLAGS "-dylib -w -lm")
+    endif()
```
Comment (isuruf) [inline CMakeLists.txt]: @franz, `dynamiclib` has to be used depending on whether clang or the linker is used in https://github.com/pocl/pocl/blob/8c36c380afdfeebf9216627fa97a829f7a3d6cd5/lib/CL/devices/common.c#L198
If `clang` is used, `-dynamiclib` has to be given and if `ld` is used `-dylib` has to be used.

**Comment by franz** [comment]
LD vs Clang should now be fixed in master.

**Pull request closed without merging**